cli/command/formatter/image.go

```diff
@@ -174,10 +178,34 @@ func imageFormat(ctx ImageContext, images []types.ImageSummary, format func(subC
 				}
 			}
 		}
-		for _, imageCtx := range images {
-			if err := format(imageCtx); err != nil {
-				return err
-			}
+	}
+
+	bys := []string{}
+	if ctx.Sortby == "" {
+		bys = append(bys, "CreatedAt:desc")
+	} else {
+		bys = append(bys, strings.Split(ctx.Sortby, ",")...)
+	}
+
+	// whitelist is a map of struct field name to new name
+	whitelist := map[string]string{
+		"Created": "CreatedAt",
+		"ID":      "",
+		"Size":    "",
+		"repo":    "Repository",
+		"tag":     "Tag",
+		"digest":  "Digest",
+	}
+	sorter, err := newGenericStructSorter(imageCtxs, bys, whitelist)
+	if err != nil {
+		return err
+	}
+
+	sort.Sort(sorter)
```

Comment:
Looks like we're now always sorting, but before this change, we didn't? I think we should only create a sorter if `ctx.SortBy` is not empty / `bys` is not empty, and otherwise print the results as-is. The results are already sorted in the daemon; https://github.com/docker/docker/blob/254fc83cba90ed79c78f4cb0cb33aeeaff492798/daemon/images.go#L230